tests/test_headers.py

```diff
@@ -117,6 +117,16 @@ def on_head(self, req, resp):
         resp.content_location = self.URL1
 
 
+class UnicodeHeaderResource:
+
+    def on_get(self, req, resp):
+        resp.set_headers([
+            (u'X-auTH-toKEN', 'toomanysecrets'),
+            ('Content-TYpE', u'application/json'),
+            (u'X-symBOl', u'\u0040'),
```

Comment:
Would you mind adding a negative test that includes a non-ascii character, and asserting that it does indeed raise an error?